File: codewars/201-300/reverse-polish-notation-calculator.java
import java.util.Stack;

/**
 * Created by ice1000 on 17-6-29.
 *
 * @author ice1000
 */
public class Calc {

		public double evaluate(String s) {
				if ("".equals(s)) return 0;
				String[] seq = s.split(" ");
				Stack<Double> stack = new Stack<>();
				for (String item : seq)
						switch (item) {
						case "+":
								stack.push(stack.pop() + stack.pop());
								break;
						case "-":	{
								double a = stack.pop();
								double b = stack.pop();
								stack.push(b - a);
						}
								break;
						case "*":
								stack.push(stack.pop() * stack.pop());
								break;
						case "/": {
								double a = stack.pop();
								double b = stack.pop();
								stack.push(b / a);
						}
								break;
						default:
								stack.push(Double.parseDouble(item));
						}
				try { return stack.lastElement(); }
				catch (Throwable e) { return 0; }
		}
}

File: codewars/301-400/reverse-polish-notation-calculator.cs
using System;
using System.Linq;
using System.Collections.Generic;

public class Calc
{
		public double evaluate(string expr)
		{
				if ("".Equals(expr)) return 0;
				var stack = new Stack<double>();
				var split = expr.Split(' ');
				foreach (var sym in split)
						try
						{
								stack.Push(double.Parse(sym));
						}
				catch (Exception)
				{
						var b = stack.Pop();
						var a = stack.Pop();
						switch (sym)
						{
								case "+":
										stack.Push(a + b);
										break;
								case "-":
										stack.Push(a - b);
										break;
								case "*":
										stack.Push(a * b);
										break;
								case "/":
										stack.Push(a / b);
										break;
						}
				}
				return stack.Pop();
		}
}

